### ISSUE TITLE
What features will Aftman have over Foreman?

### ISSUE THREAD

**Issue by Dekkonot** [LPGhatguy/aftman#1]
Other than the obvious benefit of not being owned by Roblox (who seem to have no interest in maintaining Foreman), what will Aftman have over Foreman?

I believe you answered this on Discord already but for posterity it should be recorded on GitHub too.

**Comment by LPGhatguy** [comment]
I should turn this into the README at some point.

I'm hoping to fix some of the core design mistakes I made in Foreman and also take a little more care with the codebase. Roughly:

* Exact version dependencies. Using a range here has tripped up lots of users and I realized at some point that this isn't great.
* Commands to install, uninstall, and upgrade tools. Editing your global, tucked-away toml file is rough.
* Change model to no longer trust-by-default. Aftman will prompt before downloading new tools. (Roblox/foreman#16).
* Different strategy for storing executables. (Roblox/foreman#11)
* Better heuristics for picking the right artifacts for your platform. Compiler, OS, architecture, plus custom patterns. (Roblox/foreman#18)
* Proper error handling. I used `unwrap` all over because Foreman was meant to be a prototype. This should make diagnosing issues a lot easier.
* Less Roblox-angled. I think that this sort of tool would be useful in general, not just for the Roblox community, so I want to avoid making it overtly Roblox.
